src/firecracker/src/api_server_adapter.rs

```diff
@@ -149,25 +149,35 @@ pub(crate) fn run_with_api(
         .name("fc_api".to_owned())
         .spawn(move || {
             match ApiServer::new(to_vmm, from_vmm, to_vmm_event_fd).bind_and_run(
-                api_bind_path,
+                &api_bind_path,
                 process_time_reporter,
                 &api_seccomp_filter,
                 api_payload_limit,
                 socket_ready_sender,
             ) {
                 Ok(_) => (),
-                Err(api_server::Error::Io(inner)) => match inner.kind() {
-                    std::io::ErrorKind::AddrInUse => panic!(
-                        "Failed to open the API socket: {:?}",
-                        api_server::Error::Io(inner)
-                    ),
-                    _ => panic!(
-                        "Failed to communicate with the API socket: {:?}",
-                        api_server::Error::Io(inner)
-                    ),
-                },
-                Err(eventfd_err @ api_server::Error::Eventfd(_)) => {
-                    panic!("Failed to open the API socket: {:?}", eventfd_err)
+                Err(api_server::Error::ServerCreation(ServerError::IOError(inner))) => {
+                    match inner.kind() {
+                        std::io::ErrorKind::AddrInUse => {
+                            let sock_path = api_bind_path.display().to_string();
+                            error!(
+                                "Failed to open the API socket at: {sock_path}. Check that it is \
+                                 not already used."
+                            );
+                            std::process::exit(vmm::FcExitCode::GenericError as i32);
+                        }
+                        _ => {
+                            error!(
+                                "Failed to bind and run the HTTP server: {}",
+                                ServerError::IOError(inner)
+                            );
+                            std::process::exit(vmm::FcExitCode::GenericError as i32);
+                        }
+                    }
+                }
+                Err(api_server::Error::ServerCreation(err)) => {
+                    error!("Failed to bind and run the HTTP server: {err}");
+                    std::process::exit(vmm::FcExitCode::GenericError as i32);
```

Comment:
nit: de-duplicate the match arms, at the expense of a slightly longer line
```suggestion
                Err(api_server::Error::ServerCreation(ServerError::IOError(inner))) if inner.kind() == std::io::ErrorKind::AddrInUse => {
                    let sock_path = api_bind_path.display().to_string();
                    error!(
                        "Failed to open the API socket at: {sock_path}. Check that it is \
                         not already used."
                    );
                    std::process::exit(vmm::FcExitCode::GenericError as i32);
                }
                Err(api_server::Error::ServerCreation(err)) => {
                    error!("Failed to bind and run the HTTP server: {err}");
                    std::process::exit(vmm::FcExitCode::GenericError as i32);
```